robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowAssetManager.java.vm

```diff
@@ -417,16 +420,28 @@ public final class ShadowAssetManager {
     AttributeResource attribute = findAttributeValue(resId, set, styleAttrStyle, defStyleFromAttr, defStyleFromRes, themeStyleSet);
     while (attribute != null && attribute.isStyleReference()) {
       ResName otherAttrName = attribute.getStyleReference();
-
-      attribute = themeStyleSet.getAttrValue(otherAttrName);
-      if (attribute != null) {
-        attribute = new AttributeResource(resourceLoader.getResourceIndex().getResName(resId), attribute.value, attribute.contextPackageName);
+      ResName resName = resourceLoader.getResourceIndex().getResName(resId);
+
+      AttributeResource otherAttr = themeStyleSet.getAttrValue(otherAttrName);
+      if (otherAttr == null) {
+        strictError("no such attr %s in %s while resolving value for %s", attribute.value, themeStyleSet, resName.getFullyQualifiedName());
+        attribute = null;
+      } else {
+        attribute = new AttributeResource(resName, otherAttr.value, otherAttr.contextPackageName);
       }
     }
 
     return attribute;
   }
 
+  private void strictError(String message, Object... args) {
```

Comment:
Maybe "strictResources" is a category we could switch on in a central place. Logger doesn't feel like that place, since its job is logging, but we don't want to have these checks spread over multiple locations. Another location we can enforce / warn of strictResource violations is the ResourceLoader where currently we have duplicate id - name mappings or inconsistencies. Its probably something we'd want to leave as a warning for a while but it would be good to have a timeline when switching to strictResources = true as the default. Maybe we can factor this into a StrictChecks class with a series of categories. Other categories could be missing Is ShadowApplication.checkActivities()